src/IteratorIr.hs

```diff
@@ -50,6 +53,13 @@ lift2 :: (Iterator b c t -> Iterator b c' t' -> r) -> Iterator a c t -> Iterator
 lift2 g (It f a) (It f' a') = It (\x -> g (It f x) (It f' x)) a  -- TODO: a and a' must be identical!
 
 
+-- This looks like a Traversal optic (see: Control.Lens.Traversal)
+-- neighborhood :: Traversal i j (Iterator i p t) (Iterator j p t)
+-- neighborhood :: Applicative f => (i -> f j) -> Iterator i p t -> f (Iterator j p t)
+neighborhood :: (i -> [j]) -> Iterator i p t -> [Iterator j p t]
+neighborhood i2j (It acc i) = fmap (It acc) (i2j i)
```

Comment:
I’d prefer type parameter and variable names that are more consistent with the code above and below here, too.